lib/seattleflu/id3c/cli/command/etl/redcap_det_childcare.py

```diff
@@ -0,0 +1,535 @@
+INTERNAL_SYSTEM = 'https://seattleflu.org'
+
+SWAB_AND_SEND_SITE = 'ChildcareSwabNSend'
+RADFORD_SITE = 'UWChildrensCenterRadfordCourt'
+SANDPOINT_SITE = 'ChildcareCenter70thAndSandPoint'
+
+ENROLLMENT_EVENT_NAME_PREFIX = 'enrollment_arm_'
+ENCOUNTER_EVENT_NAME_PREFIX = 'week'
+COLLECTION_CODE = CollectionCode.HOME_HEALTH
+
+
+# Don't include baseline_screening because it's completed by a staff member,
+# not the participant.
+REQUIRED_ENROLLMENT_INSTRUMENTS = [
+    'consentassent_form',
+    'enrollment_questionnaire'
+]
+
+
+def command_for_each_project(function):
+    """
+    A decorator to register one redcap-det subcommand per REDCap project, each
+    calling the same base *function*.
+    Used for side-effects only; the original *function* is unmodified.
+    """
+    for project in PROJECTS:
+        help_message = 'Process REDCap DETs for the Childcare Study'
+
+        redcap_det.command_for_project(
+            name = project.command_name,
+            redcap_url = REDCAP_URL,
+            project_id = project.id,
+            raw_coded_values = True,
+            revision = REVISION,
+            help = help_message,
+            include_incomplete = True)(function)
+
+    return function
+
+
+@command_for_each_project
+def redcap_det_childcare(*, db: DatabaseSession, cache: TTLCache, det: dict,
+    redcap_record_instances: List[REDCapRecord]) -> Optional[dict]:
+
+    assert redcap_record_instances is not None and len(redcap_record_instances) > 0, \
+        'The redcap_record_instances list was not populated.'
+
+    enrollments = [record for record in redcap_record_instances if \
+        record.event_name.startswith(ENROLLMENT_EVENT_NAME_PREFIX)]
+    assert len(enrollments) == 1, \
+        f'Record had {len(enrollments)} enrollments.'
+
+    enrollment = enrollments[0]
+
+    incomplete_enrollment_instruments = {
+                instrument
+                    for instrument
+                    in REQUIRED_ENROLLMENT_INSTRUMENTS
+                    if not is_complete(instrument, enrollment)
+            }
+
+    if incomplete_enrollment_instruments:
+        LOG.debug(f'The following required enrollment instruments «{incomplete_enrollment_instruments}» are not yet marked complete.')
+        return None
+
+    # If the participant's age < 18 ensure we have parental consent.
+    if (enrollment['core_age_years'] == "" or int(enrollment['core_age_years']) < 18) and \
+        enrollment['parent_signature'] == '':
+        LOG.debug("The participant is < 18 years old and we do not have parental consent. Skipping record.")
+        return None
+
+    # Create the participant resource entry and reference.
+    # Assumes that the project language is the participant's preferred language.
+    patient_entry, patient_reference = create_patient_using_demographics(
+        sex = 'unknown', # Set to unknown so that we don't ingest identifiers
+        preferred_language = LANGUAGE_CODE[enrollment.project.id],
```

Comment:
Good point. It would be helpful to tell the staff that the PT prefers Spanish. I made the change in the "FIXUP: jover's feedback" commit.

Reply:
Also, I'm making a change to _create_patient to check whether preferred_language is populated. Otherwise, we'd have a communication entry that didn't have a language set.